src/SFML/Audio/SoundFileReaderMp3.cpp

```diff
@@ -68,14 +68,14 @@ namespace
 std::size_t readCallback(void* ptr, std::size_t size, void* data)
 {
     auto* stream = static_cast<sf::InputStream*>(data);
-    return static_cast<std::size_t>(stream->read(ptr, static_cast<std::int64_t>(size)));
+    return stream->read(ptr, size).value_or(-1);
 }
 
 int seekCallback(std::uint64_t offset, void* data)
 {
-    auto*              stream   = static_cast<sf::InputStream*>(data);
-    const std::int64_t position = stream->seek(static_cast<std::int64_t>(offset));
-    return position < 0 ? -1 : 0;
+    auto*      stream   = static_cast<sf::InputStream*>(data);
+    const auto position = stream->seek(static_cast<std::size_t>(offset));
+    return position ? 0 : -1;
```

Comment:
```suggestion
    return position.has_value() ? 0 : -1;
```

Reply:
Same here, another possible idea is changing the name to `maybePosition` or `optionalPosition` but that's my least favorite option of everything I've suggested.